[KSanjith] iP

KSanjith opened the pull request.

KSanjith committed:
Rename chatbot to Doug
Add greet and exit skeleton code

KSanjith committed:
Let users add text and list added text
Modify personality of chatbot

KSanjith committed:
Add ability to mark/unmark tasks as done
Create new Task class to store tasks

KSanjith committed:
Tweak the code to comply with coding standards

KSanjith committed:
Fix boundary checking for mark/unmark index

KSanjith committed:
Add ToDos, Deadlines and Events
Neaten code in Main

KSanjith committed:
Set up automated testing
Fix critical bug of creating multiple Scanners

KSanjith committed:
Improve code quality
Remove magic literals
Structure code logically
Meet character limit per line
Refactor code (extract method)

limkongkiat commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+
+public class Doug {
+
+    private static final String dashedLine = "______________________________________________\n";
```
Constants should be named with all caps (i.e. DASHED_LINE)

limkongkiat commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+    }
+
+    public static void doToDo(String command) {
+        Todo t = new Todo(command.substring(5));
```
Variable could be more clearly named (e.g. newTask instead of t)

limkongkiat replied:
A clearer variable name could be used (e.g. newTask instead of t)

limkongkiat commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+                }
+                doToDo(command);
+            } else if (command.startsWith("deadline")) {
+                if (counter >= 100) {
```
Max. number of tasks should be labelled as a constant instead of using a magic number (100) here

naveen42x replied:
Would it be better to change this to a switch case instead of using an if else statement?

limkongkiat commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+                "Hope to see you around these parts again soon!\n" + dashedLine);
+
+    }
+}
```
LGTM, overall good adherence to Coding standards and clear use of constants instead of using magic literals

limkongkiat reviewed (commented):
Overall good job!

Progresst-8 commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+            } else {
+                System.out.println(dashedLine + "Something seems off partner...\n" + dashedLine);
+            }
+        }
```
Should the big if-else block be a switch-case instead?

Progresst-8 commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+
+    static final int EVENT_NAME_START = 6;
+    static final int EVENT_FROM_START = 6;
+    static final int EVENT_TO_START = 4;
```
Can I ask why you decided to hardcode the string parsing for deadlines and events?

naveen42x replied:
Could the purpose of this be explained?

Progresst-8 commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+        if (listIndex > counter || listIndex <= 0) {
+            System.out.println(dashedLine + "No can do bud, your input is invalid!\n" + dashedLine);
+            return;
+        }
```
Should you extract the index-within-range checker in the methods doMark and doUnmark?

Progresst-8 commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+                if (counter >= 100) {
+                    System.out.println(dashedLine + "Sorry partner, the list is full!\n" + dashedLine);
+                    continue;
+                }
```
Should you extract the list-is-full checker into its own function? I see there is some repetition (for deadline, event, todo) of checking over 100 and printing to output that the list is full.

Progresst-8 reviewed (commented):
LGTM overall. The code is generally readable, though there are some repeated chunks of code that could be refactored into its own methods.
Keep up the good work!

naveen42x commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+
+    private static final String dashedLine = "______________________________________________\n";
+    private static int counter = 0;
+    private static Task[] tasks = new Task[100];
```
Would it be better to make the magic number a named constant?

naveen42x commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+        int indexOfSlash = command.indexOf('/');
+
+        String taskName = command.substring(DEADLINE_NAME_START, indexOfSlash - 1);
+        String taskDeadline = command.substring(indexOfSlash + DEADLINE_DATE_START);
```
Avoid using hard-coded values

naveen42x commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+
+        String taskName = command.substring(EVENT_NAME_START, indexOfFirstSlash - 1);
+        String taskStart = command.substring(indexOfFirstSlash + EVENT_FROM_START, indexOfSecondSlash - 1);
+        String taskEnd = command.substring(indexOfSecondSlash + EVENT_TO_START);
```
Avoid using hard-coded values

naveen42x commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+
+        Event e = new Event(taskName, taskStart, taskEnd);
+        tasks[counter] = e;
+        counter++;
```
Could there be a check to ensure that there’s enough space in the array before assigning values?

naveen42x commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+        String taskDeadline = command.substring(indexOfSlash + DEADLINE_DATE_START);
+
+        Deadline d = new Deadline(taskName, taskDeadline);
+        tasks[counter] = d;
```
Could use a more descriptive and clear variable name (e.g. deadlineTask instead of d)

naveen42x commented on src/main/java/Doug.java:
```diff
@@ -0,0 +1,180 @@
+        String taskEnd = command.substring(indexOfSecondSlash + EVENT_TO_START);
+
+        Event e = new Event(taskName, taskStart, taskEnd);
+        tasks[counter] = e;
```
Could use a more descriptive and clear variable name (e.g. eventTask instead of e)

naveen42x reviewed (commented):
Overall good quality. Variable naming could be more descriptive in general.

KSanjith committed:
Adopt suggestions from PR reviews

KSanjith committed:
Add DougException class for exception handling
Modify hard-coded parts of code to be dynamic

KSanjith committed:
Merge branch 'branch-Level-5'
Handle errors using exceptions

KSanjith committed:
Divide classes into 2 Packages

KSanjith committed:
Merge branch 'branch-A-Packages'

KSanjith committed:
Modify testing command to use new directories

KSanjith committed:
Use dynamic arraylist instead of fixed size array
Remove magic numbers from mark and unmark methods

KSanjith committed:
Allow user to delete items from the list

KSanjith committed:
Save and load tasks from .txt file

KSanjith committed:
Merge branch 'branch-Level-6'

KSanjith committed:
Merge branch 'branch-Level-7'

KSanjith committed:
Delete will now delete from .txt file too

KSanjith committed:
Create .jar file
Add *.jar to gitignore

Kobot7 commented on src/main/java/doug/command/Doug.java:
```diff
@@ -0,0 +1,420 @@
+            tasks.get(listIndex - 1).markAsDone();
+            try {
+                saveTasks();
+            } catch (IOException e) {
```
Is the variable name "e" informative enough?

Kobot7 commented on src/main/java/doug/command/Doug.java:
```diff
@@ -0,0 +1,420 @@
+
+import java.util.Scanner;
+import java.util.ArrayList;
+import doug.tasks.*;
```
Avoid using universal imports

Kobot7 commented on src/main/java/doug/command/Doug.java:
```diff
@@ -0,0 +1,420 @@
+        System.out.println("Your list is now " + counter + " tasks long partner\n" + DASHED_LINE);
+    }
+
+    public static void doEvent(String command) throws DougException {
```
Method is a little long, consider extracting code out to sub-methods instead

Kobot7 commented on src/main/java/doug/command/Doug.java:
```diff
@@ -0,0 +1,420 @@
+        }
+
+        int indexOfFirstBoundary = line.indexOf("| ");
+        String taskName = line.substring(0, indexOfFirstBoundary).trim();
```
Consider using a named constant instead of a magic number

Kobot7 commented on src/main/java/doug/command/Doug.java:
```diff
@@ -0,0 +1,420 @@
+        System.out.print(DASHED_LINE);
+    }
+
+    public static void doMark(String command) throws DougException {
```
Naming of methods could be clearer. Using 2 verbs isn't very informative. You are "do"-ing "mark"-ing, but marking what?

KSanjith committed:
Add Ui, Storage, Parser, TaskList, Command classes

KSanjith committed:
Add find command to search existing tasks

KSanjith committed:
Store, print and save dates and times

KSanjith committed:
Add JavaDoc header comments to all non-private
classes and methods

KSanjith committed:
Merge pull request #1 from KSanjith/branch-Level-8

Store, print and save dates and times

KSanjith committed:
Merge branch 'master' into branch-Level-9

KSanjith committed:
Merge branch 'master' into branch-A-JavaDoc

KSanjith committed:
Merge pull request #2 from KSanjith/branch-Level-9

Add find command to search existing tasks

KSanjith committed:
Merge branch 'master' into branch-A-JavaDoc

KSanjith committed:
Merge pull request #3 from KSanjith/branch-A-JavaDoc

Add JavaDoc header comments to all non-private classes and methods

KSanjith committed:
Make minor fixes and improvements
Add JavaDoc header to all task classes
Add exception catching for non-integer index
inputs for mark, unmark and delete commands

KSanjith committed:
Start working on README

KSanjith committed:
Instantiate UI and Storage objects in main
Instead of directly accessing static methods

KSanjith committed:
Finish writing README
Change Doug class to resemble suggested format
Modify code to use create more objects
Add more JavaDoc headers and comments

KSanjith committed:
Fix formatting issues in README.md

KSanjith committed:
Make more formatting fixes for README.md

KSanjith committed:
Change image formatting in README.md

KSanjith committed:
Create a help command

KSanjith committed:
Fix bug where todo task can be added without name

KSanjith committed:
Add check for the right order of /from and /to
Fix whitespace bug in loading /from of event task

KSanjith committed:
Allow inputs to have same name as commands

KSanjith committed:
Package the UI, Storage, Parser, TaskList classes
Fix some Javadoc headers

KSanjith committed:
Final Release
Add full stops to ends of all Javadoc comments
